community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreAbstractRelationshipCursor.java

```diff
@@ -108,15 +114,28 @@ public long otherNode( long nodeId )
                 relationshipRecord.getSecondNode() : relationshipRecord.getFirstNode();
     }
 
+    private Lock shortLivedReadLock()
+    {
+        Lock lock = lockService.acquireRelationshipLock( relationshipRecord.getId(), LockService.LockType.READ_LOCK );
+        if ( lockService != NO_LOCK_SERVICE )
+        {
+            // It's safer to re-read the relationship record here, specifically nextProp, after acquiring the lock
+            neoStores.getRelationshipStore().fillRecord( relationshipRecord.getId(), relationshipRecord, FORCE );
```

Comment:
same thing here, if this could throw, perhaps we should guard and release the lock.